### PR TITLE
AMD SOF Soundwire generic machine driver code refactor

### PR DESCRIPTION
**Pull request by vijendarmukunda**

### CODE REVIEW

**Review by plbossart — DISMISSED** [review]
looks ok to me but @charleskeepax and @bardliao must review

**Comment by vijendarmukunda** [comment]
@plbossart @bardliao @charleskeepax : could you please review ?

File: sound/soc/sdw_utils/soc_sdw_utils.c [hunk 1]
```diff
@@ -1005,5 +1005,168 @@ int asoc_sdw_init_simple_dai_link(struct device *dev, struct snd_soc_dai_link *d
...
+
+	INIT_LIST_HEAD(&dailinks->endpoints);
+	dailinks->group_id = new->group_id;
+	dailinks->initialised = true;
+
+	return dailinks;
+}
+EXPORT_SYMBOL_NS(asoc_sdw_find_dailink, SND_SOC_SDW_UTILS);
+
+static const char * const type_strings[] = {"SimpleJack", "SmartAmp", "SmartMic"};
```
Comment (charleskeepax) [inline sound/soc/sdw_utils/soc_sdw_utils.c]: This is only used in the debug statement below, I would be tempted to drop it and just print out the number directly. The trouble with having the strings here is there ends up being an expectation that this array, the array in the AMD stuff, and the array in the Intel stuff all match. Simpler to just avoid the problem and allow Intel/AMD to have control over what names are used on their systems. Either that or if we do want to make them all the same export the array and share it, but I think I prefer the former.

**Review by charleskeepax — CHANGES_REQUESTED** [review]
Mostly looks good one very minor comment from me.

File: sound/soc/sdw_utils/soc_sdw_utils.c (hunk 1 above)
Comment (vijendarmukunda) [reply sound/soc/sdw_utils/soc_sdw_utils.c]: Will drop type_strings from the debug statement in asoc_sdw_parse_sdw_endpoints() function  , will print only dai type.
will append AMD tag for type_strings used in AMD machine driver code.

**Comment by vijendarmukunda** [comment]
@charleskeepax : dropped type strings from debug statement. There is no hard requirement to change type_strings in AMD code as of now. Changing the type_strings in driver code, also require changes in topology file. For current platform , we want to go with no modifications in type_strings for AMD source code.

**Comment by vijendarmukunda** [comment]
@bardliao : please review the PR

**Review by charleskeepax — APPROVED** [review]
Yeah no need to change the strings. But this way there is less coupling between the two drivers. Anyways all looks good to me.

**Review by bardliao — APPROVED** [review]
LGTM

**Comment by vijendarmukunda** [comment]
@bardliao @plbossart : as PR got approved could you please merge the PR? we have other patches to be submitted on top of it?

**Pull request merged**